### ISSUE TITLE
`ExpressionBodySyntax` false positive when more than one return statement

### ISSUE THREAD

**Issue by haluzpav** [detekt/detekt#7829]
## Expected Behavior
When function has more than one return statement, `ExpressionBodySyntax` should not be triggered.

## Observed Behavior
It is.

## Steps to Reproduce
Example code:

```kotlin
class PromoCodeSummaryMapper {
    fun PromoApplied.toPresentable(currency: Currency): PromoCodeSummary? {
        return PromoCodeSummary(
            code = code,
            type = when {
                promoFreeDelivery == true -> PromoCodeSummary.FreeDelivery
                promoDiscountAmount != null -> PromoCodeSummary.Discount(Amount(-promoDiscountAmount, currency))
                else -> return null
            },
        )
    }
}
```

> .../PromoCodeSummaryMapper.kt:11:9: Functions with exact one statement, the return statement, can be rewritten with ExpressionBodySyntax. [ExpressionBodySyntax]

When I do as suggested by the issue, I get IDE error:

<img width="846" alt="Screenshot 2024-12-10 at 10 22 02" src="https://github.com/user-attachments/assets/7cbfd543-86c1-4299-bc68-f416b1a49ebf">

## Your Environment

* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): 8.10
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by maxweilun1989** [comment]
@BraisGabin  Hi, I would like to give it a try. Could you assign it to me?
